dist/Array.js

```diff
@@ -1,112 +1,157 @@
-        },
-    },
-    findMap: {
-        enumerable: false,
-        writable: true,
-        configurable: true,
-        value: function (predicate, map) {
-            var found = this.find(predicate);
-            return found !== undefined ? map(found) : found;
-        },
-    },
+	remove: {
+		enumerable: false,
+		writable: true,
+		configurable: true,
+		value: function (elem) {
+			// do not use filter to modify current array
+			var index = this.indexOf(elem);
+			if (index === -1) return this;
+			this.splice(index, 1);
+			return this; //.filter((e) => e !== elem);
+		},
+	},
+	insert: {
+		enumerable: false,
+		writable: true,
+		configurable: true,
+		value: function (elem, index) {
+			if (!index) index = this.length;
+			return this.splice(index, 0, elem);
+		},
+	},
+	flat: {
+		enumerable: false,
+		writable: true,
+		configurable: true,
+		value: function (depth) {
+			if (depth === void 0) {
+				depth = 1;
+			}
+			return this.reduce(function (acc, val) {
+				return (Array.isArray(val) && depth >= 1) || depth === -1
+					? acc.concat(val.flat(depth--))
+					: acc.concat(val);
+			}, []);
+		},
+	},
+	last: {
+		enumerable: false,
+		writable: true,
+		configurable: true,
+		value: function () {
+			return this[this.length - 1];
+		},
+	},
+	first: {
+		enumerable: false,
+		writable: true,
+		configurable: true,
+		value: function () {
+			return this[0];
+		},
+	},
+	unique: {
+		enumerable: false,
+		writable: true,
+		configurable: true,
+		value: function () {
+			return __spread(new Set(this));
+		},
+	},
+	random: {
+		enumerable: false,
+		value: function () {
+			return this[Math.floor(Math.random() * this.length)];
+		},
+	},
+	shuffle: {
+		enumerable: false,
+		writable: true,
+		configurable: true,
+		value: function () {
+			var _a;
+			for (var i = this.length - 1; i > 0; i--) {
+				var j = Math.floor(Math.random() * (i + 1));
+				(_a = __read([this[j], this[i]], 2)), (this[i] = _a[0]), (this[j] = _a[1]);
+			}
+			return this;
+		},
+	},
+	findMap: {
+		enumerable: false,
+		writable: true,
+		configurable: true,
+		value: function (predicate, map) {
+			var found = this.find(predicate);
+			return found !== undefined ? map(found) : found;
+		},
+	},
+	count: {
+		enumerable: false,
+		writable: true,
+		configurable: true,
+		value: function (search) {
+			const nativeTypes = [String, Number, Object, Array];
+			let i = 0;
+			this.forEach((element) => {
+				if (element === search) {
+					i++;
+					return;
+				}
+				// check if searchparam is a native class (l: 99)
+				if (typeof search == "function") {
+					if (
+						nativeTypes.includes(search) &&
+						typeof element === search.toString().split(" ")[1].slice(0, -2).toLowerCase()
+					) {
+						i++;
+						return;
+					} else if (element instanceof search) {
+						i++;
+						return;
+					}
+				}
+				// if element of array is a string: allow regex patterns
+				if (typeof element === "string") {
+					if (element.match(search)) {
```

Comment:
will fail if search is not a regex

Reply:
yes, as intended.
One can either provide a Class, a regex pattern or anyother type, meaning if you search for a number it will find that number or none, this works as intended.